packages/cletus/src/operations/clerk.tsx

```diff
@@ -665,6 +666,148 @@ export const file_read = operationOf<
   ),
 });
 
+// Helper function to generate file edit content and diff
+async function generateFileEditDiff(
+  input: { path: string; request: string; offset?: number; limit?: number },
+  fileContent: string,
+  ai: any
+): Promise<{ newFileContent: string; diff: string; changed: boolean }> {
+  // Paginate the text if offset/limit are provided (lines only)
+  const paginatedContent = paginateText(fileContent, input.limit, input.offset, 'lines');
+
+  // Use AI to generate new content based on the request
+  const models = ai.config.defaultContext!.config!.getData().user.models;
```

Comment:
Lets add another model to models called `edit` and update all the places models are configured/specified. We first look at edit model then use chat as backup.